### PR TITLE
[IA-3386] Prepare OUCRC API

### PR DESCRIPTION
**Pull request by tdethier**
Adds a web + a mobile API for `OrgUnitChangeRequestConfiguration`

Related JIRA tickets : IA-3386, IA-2896 (epic)

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [x] Are there enough tests
- [x] Documentation has been included (for new feature)

## Doc
Documentation in: 
- the code
- the `docs` folder 
- the Jira Epic 
- Whimsical (see Jira Epic)

## Changes

- Added 2 new endpoints in API
  - Web - `api/orgunits/changes/configs/`
      - GET - list
      - GET - retrieve
      - GET - check_availability
      - POST
      - PATCH
      - DELETE
  - Mobile -  `api/mobile/orgunits/changes/configs/`
    - GET - list
- Added tests for both endpoints

## How to test

Go to the web API, try the various available calls.

(If there are some issues with many-to-many relationships during creation, it's because of how DRF sends m2m data, but it works fine if you create OUCRCs through the Django admin panel, or if you send a payload with curl/postman/....)

## Print screen / video

image

## Notes
/

### CODE REVIEW

File: iaso/api/org_unit_change_request_configurations/views.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    OrgUnitChangeRequestConfigurationWriteSerializer,
+    OrgUnitChangeRequestConfigurationUpdateSerializer,
+    OrgUnitTypeNestedSerializer,
+    OrgUnitChangeRequestConfigurationAuditLogger,
+    ProjectIdSerializer,
+)
+from iaso.models import OrgUnitChangeRequestConfiguration, OrgUnitType, Project
+
+
+class OrgUnitChangeRequestConfigurationViewSet(viewsets.ModelViewSet):
```
Comment (mathvdh) [inline iaso/api/org_unit_change_request_configurations/views.py:21]: Is there a specific reason you are not deriving from iaso.api.commom.ModelViewSet ?

**Review by mathvdh — CHANGES_REQUESTED** [review]
At the moment there is no way to test the mobile using project's swagger : Can you add a `@swagger_auto_schema(manual_parameters=[app_id_param])` (if you do a search `app_id_param` in the project you will find it)

Screenshot 2024-09-23 at 10 49 25

File: iaso/api/org_unit_change_request_configurations/views.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        "org_units_editable",
+        "created_at",
+        "updated_at",
+        "created_by__username",
+        "updated_by__username",
+    ]
+    http_method_names = ["get", "post", "patch", "delete"]
+    pagination_class = OrgUnitChangeRequestConfigurationPagination
+
+    def get_queryset(self):
```
Comment (mathvdh) [inline iaso/api/org_unit_change_request_configurations/views.py:46]: I think we don't want to allow a user to see all the OUCRC of all the accounts from the API.
This should be filtered by the account of the user and project.account and maybe by user.iaso_profile.project ? Not sure about the second one ...

Comment (mathvdh) [inline iaso/api/org_unit_change_request_configurations/views.py:46]: I think we don't want to allow a user to see all the OUCRC of all the accounts from the API.
This should be filtered by the account of the user and project.account and maybe by user.iaso_profile.project ? Not sure about the second one ...

Comment (mathvdh) [reply iaso/api/org_unit_change_request_configurations/views.py:46]: At the moment when you query `/api/orgunits/changes/configs/ `it returns even OUCRC which I'm not supposed to see as they are link to another project of another account ...

Comment (mathvdh) [reply iaso/api/org_unit_change_request_configurations/views.py:46]: if you would change the `get_queryset` that would correct the fact that at the moment you can delete a OUCRC from another project/account (I think..)

File: iaso/api/org_unit_change_request_configurations/views.py (hunk 1 above)
Comment (tdethier) [reply iaso/api/org_unit_change_request_configurations/views.py:21]: I was told that the `OrgUnitChangeRequest` API was the most up to date and to do something similar to this one. It doesn't inherit from `iaso.api.common.ModelViewSet` :man_shrugging:

File: iaso/api/org_unit_change_request_configurations/views.py (hunk 2 above)
Comment (tdethier) [reply iaso/api/org_unit_change_request_configurations/views.py:46]: Yes, there should be some filtering done here. I didn't add a comment there, but I left all permission/authorization checks for another ticket

**Comment by tdethier** [comment]
@mathvdh I added the swagger annotation you wanted

**Review by mathvdh — APPROVED** [review]
👍🏼

**Pull request merged**